plugins/arm-lifter/armng_env.ml

```diff
@@ -0,0 +1,101 @@
+open Bap_core_theory
+open Base
+open KB.Syntax
+
+module Env = struct
+  type value
+  type byte
+  type reg_single
+  type half_byte
+  type half_word
+  type bit_val
+  type double_word
+
+  let bit : Theory.Bool.t Theory.Value.sort =
+    Theory.Bool.t
+
+  let bit_val : bit_val Theory.Bitv.t Theory.Value.sort =
+    Theory.Bitv.define 1
+
+  let double_word : double_word Theory.Bitv.t Theory.Value.sort =
+    Theory.Bitv.define 64
+
+  (** grps are defined by 3-bit indices *)
+  let reg_single : reg_single Theory.Bitv.t Theory.Value.sort =
+    Theory.Bitv.define 3
+
+  let half_word : half_word Theory.Bitv.t Theory.Value.sort =
+    Theory.Bitv.define 16
+
+  let value : value Theory.Bitv.t Theory.Value.sort =
+    Theory.Bitv.define 32
+
+  let half_byte : half_byte Theory.Bitv.t Theory.Value.sort =
+    Theory.Bitv.define 4
+
+  let byte : byte Theory.Bitv.t Theory.Value.sort =
+    Theory.Bitv.define 8
+
+  let heap_type = Theory.Mem.define value byte
+
+  let memory = Theory.Var.define heap_type "mem"
+
+  (** define grps *)
+  let r0 = Theory.Var.define value "r0"
```

Comment:
please, synchronize the register names with what we had in the old lifter, i.e., capitalize, although I also dislike the caps, let's keep the tradition (the real reason, we may break a lot of downstream analysis and make comparison with the old versions hard)